unlock-app/src/components/interface/locks/Manage/index.tsx

```diff
@@ -20,21 +20,70 @@ import { useConfig } from '~/utils/withConfig'
 import { Container } from '../../Container'
 import { RiPagesLine as PageIcon } from 'react-icons/ri'
 import { FilterBar } from './elements/FilterBar'
+import { buildCSV } from '~/utils/csv'
+import FileSaver from 'file-saver'
+import { FaFileCsv as CsvIcon } from 'react-icons/fa'
+import { useStorageService } from '~/utils/withStorageService'
+import { useWalletService } from '~/utils/withWalletService'
 import { useLockManager } from '~/hooks/useLockManager'
 import { addressMinify } from '~/utils/strings'
 
 interface ActionBarProps {
   lockAddress: string
+  network: number
 }
 
 interface TopActionBarProps {
   lockAddress: string
   network: number
 }
 
-const ActionBar = ({ lockAddress }: ActionBarProps) => {
+const defaultCols = [
+  'token',
+  'lockName',
+  'expiration',
+  'keyholderAddress',
+  'lockAddress',
+  'checkedInAt',
+  'email',
+]
+
+export function downloadAsCSV(cols: string[], metadata: any[]) {
+  const csv = buildCSV(cols, metadata)
+
+  const blob = new Blob([csv], {
+    type: 'data:text/csv;charset=utf-8',
+  })
+  FileSaver.saveAs(blob, 'members.csv')
+}
+
+const ActionBar = ({ lockAddress, network }: ActionBarProps) => {
+  const { account } = useAuth()
   const [isOpen, setIsOpen] = useState(false)
-  const { network } = useAuth()
+  const storageService = useStorageService()
+  const walletService = useWalletService()
+
+  const getMembers = async () => {
+    await storageService.loginPrompt({
+      walletService,
+      address: account!,
+      chainId: network,
+    })
+    return storageService.getKeys({
+      lockAddress,
+      network,
+      filters: {
+        query: '',
+        filterKey: 'owner',
+        expiration: 'all',
+      },
+    })
+  }
+
+  const onDownloadCsv = async () => {
+    const members = await getMembers()
+    downloadAsCSV(defaultCols, members)
```

Comment:
Wait we need the user to be able to download _all_ metadata, not just the ones in `defaultCols`.

Reply:
fixed, cols are now taken from members list